terminator-commons/src/main/java/io/github/algomaster99/terminator/commons/fingerprint/JdkIndexer.java

```diff
@@ -0,0 +1,85 @@
+package io.github.algomaster99.terminator.commons.fingerprint;
+
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.Resource;
+import io.github.classgraph.ScanResult;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Set;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import nonapi.io.github.classgraph.classpath.SystemJarFinder;
+
+/**
+ * The JdkIndexer class provides a utility to list all JDK classes by scanning the JDK used for the execution of the application.
+ */
+public class JdkIndexer {
+
+    /**
+     * Returns a list of all JDK classes. The list is populated by scanning the JDK used for the execution of this application.
+     *
+     * @return a list of all JDK classes, never null.
+     */
+    public static List<JdkClass> listJdkClasses() {
+        List<JdkClass> jdkClasses = new ArrayList<>();
+        try (ScanResult scanResult = new ClassGraph()
+                .enableSystemJarsAndModules()
+                .acceptLibOrExtJars()
+                .acceptPackages("java.*", "jdk.*", "sun.*", "oracle.*", "javax.*")
```

Comment:
Can you explain why this API call is needed? Based on the above two API calls in the chain, I would assume it would load all JDK classes. This API feels like a filter.

Reply:
The calls above only enable scanning of the JDK. Without this line you only get results from the classpath